subnode/ReachabilityAnnouncer.c

```diff
@@ -467,11 +474,14 @@ void ReachabilityAnnouncer_updatePeer(struct ReachabilityAnnouncer* ra,
 static void onReplyTimeout(struct ReachabilityAnnouncer_pvt* rap, struct Address* snodeAddr)
 {
     // time out -> re-integrate the content of the message onWire into unsent
+    // only when the onWire message not out of date
     struct Message* mow = rap->msgOnWire;
     rap->msgOnWire = NULL;
-    struct Announce_Peer* p;
-    for (p = Announce_Peer_next(mow, NULL); p; p = Announce_Peer_next(mow, p)) {
-        updatePeer(rap, p, 0);
+    if (rap->timeOfLocalUpdate < rap->msgOnWireSentTime) {
+        struct Announce_Peer* p;
+        for (p = Announce_Peer_next(mow, NULL); p; p = Announce_Peer_next(mow, p)) {
+            updatePeer(rap, p, 0);
```

Comment:
Thanks for making this PR. I'm not 100% sure that it won't still blow up on occasion but if it fixes the problem for you, I guess that's good reason to merge it.
However, if I understand your diagnosis of the problem well, wouldn't it work to change this line to:

```c
struct Announce_Peer* lp = peerFromLocalState(rap->localState, p->ipv6);
if (lp) { updatePeer(rap, lp, 0); }
```
?

Let me know what you think, I'll pull your fix if you think it is better.